### DISCUSSION TITLE
Inconsistent PR permissions between REST API and GraphQL

### DISCUSSION THREAD

**Discussion by hkdobrev** [community/community#112168] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

When having the `pull_requests:read` permission, a GitHub App authenticated with access to a private repo could successfully call both the REST API and the GraphQL API to query pull requests.

Ideally, it would be able to read metadata about files and commits in the pull requests without accessing the contents of the files themselves.

However, there's a very troublesome inconsistency in the data access with different APIs.

|| PR Files metadata | PR Commits metadata |
|-|-|-|
| REST API | ✅ [files](https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-pull-requests-files) | ✅ [commits](https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-commits-on-a-pull-request) |
| GraphQL | ✅ [files](https://docs.github.com/en/graphql/reference/objects#pullrequestchangedfileconnection) | ❌ [commits](https://docs.github.com/en/graphql/reference/objects#pullrequestcommitconnection) |

Reading file names and stats is possible on both APIs without additional access.

But when reading commit dates, only the REST API provides them. GraphQL doesn't have access.

Example query:

```graphql
query TestPermissionsQuery($filesLimit: Int = 100) {
  repository(name: "private", owner: "private") {
    pullRequests(first: 100, states: [OPEN]) {
      nodes {
        files(first: $filesLimit) {
          nodes {
            additions
            deletions
            path
          }
          totalCount
        }
        firstCommit: commits(first: 1) {
          nodes {
            commit {
              committedDate
            }
          }
        }
      }
    }
  }
}
```

The files are returned, but the commits metadata is not accessible on private repos.
The same behaviour is reproduced when finding PRs via `search()` as well.

**Comment by MostafaWaleed0** [comment]
Overview of how to implement this approach:
```js
query PullRequestsQuery($filesLimit: Int = 100) {
  repository(owner: "owner", name: "repo") {
    pullRequests(first: 100, states: [OPEN]) {
      nodes {
        number
        files(first: $filesLimit) {
          nodes {
            path
            additions
            deletions
          }
          totalCount
        }
        commits(first: 1) {
          nodes {
            commit {
              committedDate
            }
          }
        }
      }
    }
  }
}
```

**Reply by hkdobrev** [reply]
@MostafaWaleed0 Are there any functional differences in the query you've outlined? The main difference I'm seeing is I have an alias for the first commit.
